python_modules/libraries/dagster-airflow/dagster_airflow/hooks/dagster_hook.py

```diff
@@ -0,0 +1,252 @@
+        super().__init__()
+        self.url = url
+        self.user_token = user_token
+        if dagster_conn_id is not None:
+            conn = self.get_connection(dagster_conn_id)
+            base_url = conn.login if conn.login else "https://dagster.cloud/"
+            if base_url == "https://dagster.cloud/":
+              self.set_hook_for_cloud(conn)
+            else:
+              self.set_hook_for_oss(conn)
+
+        if self.user_token is None:
+            raise AirflowException('Cannot get user_token: No valid user_token or dagster_conn_id supplied.')
+
+        if self.url is None:
+            raise AirflowException('Cannot get dagster url: No valid url or dagster_conn_id supplied.')
+
+    def set_hook_for_cloud(self, conn: Connection):
+      self.organization_id = conn.description
+      self.deployment_name = conn.host
+      self.user_token = conn.get_password()
+      base_url = conn.login if conn.login else "https://dagster.cloud/"
+      if self.organization_id is None or self.deployment_name is None:
+        raise AirflowException("Dagster Cloud connection requires organization_id and deployment_name to be set")
+      self.url = f"{base_url}{self.organization_id}/{self.deployment_name}/graphql"
+
+    def set_hook_for_oss(self, conn: Connection):
+      self.url = conn.login
+
+    def launch_run(
+        self,
+        repository_name:str = "my_dagster_project",
+        repostitory_location_name:str = "example_location",
+        job_name:str = "all_assets_job",
+        run_config: Optional[Dict[str, Any]] = {},
+    ) -> str:
+        query = """
+mutation LaunchJobExecution($executionParams: ExecutionParams!) {
+  launchPipelineExecution(executionParams: $executionParams) {
+    __typename
+    ... on LaunchRunSuccess {
+      run {
+        id
+        __typename
+      }
+      __typename
+    }
+    ... on PipelineNotFoundError {
+      message
+      __typename
+    }
+    ... on InvalidSubsetError {
+      message
+      __typename
+    }
+    ... on RunConfigValidationInvalid {
+      errors {
+        message
+        __typename
+      }
+      __typename
+    }
+    ...PythonErrorFragment
+  }
+}
+
+fragment PythonErrorFragment on PythonError {
+  __typename
+  message
+  stack
+  causes {
+    message
+    stack
+    __typename
+  }
+}
+        """
+        variables = {
+            "executionParams":{
+                "runConfigData": json.dumps(run_config),
+                "selector":{
+                    "repositoryName": repository_name,
+                    "repositoryLocationName": repostitory_location_name,
+                    "jobName": job_name
+                },
+                "mode":"default",
+                "executionMetadata":{
+                    "tags":[
+                        {"key":"dagster/solid_selection","value":"*"}
+                    ]
```

Comment:
do we need this tag? -- not specifying anything results in "execute all" as well. if we do need it, i thought we've updated it to op_selection cc @dpeng817